File: oi.js
async function getOI() {
  const index = "banknifty";
  const num = "23"; // for banknifty num="23"
  const expiry = "2023-09-20";
  const CEPE = "PE";
  const strikePrice = "46100.00";
  let OI_long = 0;

  const url = `https://www.moneycontrol.com/india/indexfutures/${index}/${num}/${expiry}/OPTIDX/${CEPE}/${strikePrice}/true`;

  try {
    const response = await fetch(url);
    
    if (response.ok) {
      const html = await response.text();
      const parser = new DOMParser();
      const doc = parser.parseFromString(html, "text/html");

      const divElement = doc.querySelector("div.FR.PA10");

      if (divElement) {
        const tableElement = divElement.querySelector("table");

        if (tableElement) {
          const cellData = [];
          const rows = tableElement.querySelectorAll("tr");

          for (const row of rows) {
            const columns = row.querySelectorAll("td");
            for (const column of columns) {
              cellData.push(column.textContent.trim());
            }
          }

          const OI_String = cellData[4].replace(/,/g, "");
          OI_long = parseInt(OI_String);
          // console.log(OI_long);
        } else {
          console.log("Table not found within the div.");
        }
      } else {
        console.log("Div element with class 'FR PA10' not found.");
      }
    } else {
      console.log("HTTP request failed with status:", response.status);
    }
  } catch (error) {
    console.error(error);
  }

  return OI_long;
}

// Call the function
let oi=1;
(async () => {
  const openInterest = await getOI();
  oi=openInterest;
  console.log("Open Interest:"+ oi);
})();

document.getElementById('demo').innerHTML=oi

File: oi.py
import requests
from bs4 import BeautifulSoup

index="banknifty"
num="23" #for banknifty num="23"
expiry="2023-09-20"
CEPE="PE"
strikePrice="46100.00"

url = "https://www.moneycontrol.com/india/indexfutures/"+ index+"/"+num+"/"+expiry+"/OPTIDX/"+CEPE+"/"+strikePrice+"/true"
response = requests.get(url)
soup = BeautifulSoup(response.text, 'html.parser')
cell_data=["a","","","","","","","",""]
i=0

# Find the div element with class "FR PA10"
div_element = soup.find('div', class_='FR PA10')

if div_element:
    # Find the table element within the div
    table_element = div_element.find('table')
    
    if table_element:
        # Process and extract data from the table as needed
        for row in table_element.find_all('tr'):
            columns = row.find_all('td')
            for column in columns:
                cell_data[i] = column.text
                i=i+1

        print(cell_data[4])    
    else:
        print("Table not found within the div.")
else:
    print("Div element with class 'FR PA10' not found.")



